src/service-worker/core.js

```diff
@@ -31,20 +44,36 @@ const TAG = 'cache-service-worker';
  * A list of blacklisted AMP versions that must never be served from
  * cache. Versions may be blacklisted if they contain a significant
  * implementation bug.
+ * @type {!Array<AmpVersion>}
  */
 const BLACKLIST = (self.AMP_CONFIG &&
     self.AMP_CONFIG[`${TAG}-blacklist`]) || [];
 
+
+/**
+ * Sets the default version for tests.
+ *
+ * @param {string} version
+ * @visibleForTesting
+ */
+export function setVersionForTesting(version) {
+  VERSION = version;
+}
+
 /**
  * Returns the version of a given versioned JS file.
  *
  * @param {string} url
- * @return {string}
+ * @return {RtvVersion}
+ * @visibleForTesting
  */
-function ampVersion(url) {
+export function rtvVersion(url) {
   // RTVs are 2 digit prefixes followed by the timestamp of the release.
   const matches = /rtv\/(\d{2}\d{13,})/.exec(url);
-  return matches ? matches[1] : VERSION;
+  return matches ?
+      matches[1] :
+      // 01 is the prod prefix.
+      `01${VERSION}`;
```

Comment:
i added it in there recently https://cdn.ampproject.org/sw.js, i didnt add the "canary" flag.